Sources/SQLKit/Query/SQLUnion.swift

```diff
@@ -0,0 +1,25 @@
+public struct SQLUnion: SQLExpression {
+    public let args: [SQLExpression]
+    public let all: Bool
+
+    public init(_ args: SQLExpression..., all: Bool = false) {
+        self.init(args, all: all)
+    }
+
+    public init(_ args: [SQLExpression], all: Bool = false) {
+        precondition(!args.isEmpty, "Empty SQLUnions are not valid.") 
+        self.args = args
+        self.all = all
+    }
+
+    public func serialize(to serializer: inout SQLSerializer) {
+        let groups = self.args.map(SQLGroupExpression.init)
+        guard let first = groups.first else { return }
+        first.serialize(to: &serializer)
+        for arg in groups.dropFirst() {
+            serializer.write(all ? " UNION ALL " : " UNION ")
+            arg.serialize(to: &serializer)
+        }
```

Comment:
Well! Turns out I made no less than _three_ mistakes originally:

1. Forgot the `separator:` label in `joined(separator:)`.
2. Forgot the compiler would not be able to infer the common `SQLExpression` type for the `forEach(_:)` closure.
3. Didn't put a space before `UNION` in the joiner 🤣.

Here's a version that both compiles and passes tests (I also made the helper struct public so I wouldn't be inconsistent with my remarks about consistency 😂):

```suggestion
        self.args
            .map { [SQLGroupExpression($0)] }
            .joined(separator: [SQLUnionJoiner(all: self.all)])
            .forEach { (item: SQLExpression) in item.serialize(to: &serializer) }
    }
}

public struct SQLUnionJoiner: SQLExpression {
    public var all: Bool
    
    public init(all: Bool) {
        self.all = all
    }
    
    public func serialize(to serializer: inout SQLSerializer) {
        serializer.write(" UNION\(self.all ? " ALL" : "") ")
```